tests/test_ensembles.py

```diff
@@ -357,23 +369,27 @@ def test_kmeans_modelweights(self):
         "matplotlib.pyplot" not in sys.modules, reason="matplotlib.pyplot is required"
     )
     def test_kmeans_rsqcutoff_with_graphs(self):
-        ds = xr.open_dataset(self.nc_file)
-
-        # use random state variable to ensure consistent clustering in tests:
-        [ids, cluster, fig_data] = ensembles.kmeans_reduce_ensemble(
-            data=ds.data, method={"rsq_cutoff": 0.9}, random_state=42, make_graph=True
-        )
+        if pytest.importorskip("sklearn", minversion="0.22"):
```

Comment:
Not familiar with this feature. Could this be written as a decorator to the test function?

Reply:
Not sure but we don't need the 'if' apparently @aulemahal tells me

Reply:
I haven't been able to find examples or ways to do this as decorators, but the line could be simply put at the top of the function (no need for `if`). It raises `pytest.Skip` if the package cannot be found with this minimum version.